extensions/amp-twitter/1.0/component.js

```diff
@@ -86,19 +103,48 @@ function BentoTwitterWithRef(
     ]
   );
 
+  const observerCb = useIntersectionObserver(({isIntersecting, target}) => {
```

Comment:
Does `useIntersectionObserver` fire true before the embed reaches the viewport?

Reply:
Hey, @adamsilverstein the `useIntersectionObserver` will not return true before it comes into the viewport. For that, we need to add support to provide the `rootMargin` property to the intersection observer and would require some changes to `viewport-observer.js` and `intersection-observer.js` as these are common pieces of code used across different components. I thought it would be better to address them in a separate follow-up PR.  Please let me know if you would like me to accommodate those changes in this PR itself.

Reply:
so to clarify: as you have this the twitter embed will not load until the tweet is actually in the viewport? To me, this feels like a degraded user experience, especially over a slower connection.

ideally we want to initiate the loading before the embed reaches the viewport so it is more likely to already be loaded by the time the user reaches it, right? 

can you look into this a bit further and define what needs to change to make that work?

Reply:
@adamsilverstein https://github.com/rtCamp/amphtml/pull/8/commits/f726fda698e6a9562822f31f6960cf0564dd6cad This commit adds the support for the rootMargin to intersection observer. So the embed will load 500px before it comes to the viewport. We can discuss the optimal value to use for the rootMargin.